lib/opal/nodes/super.rb

```diff
@@ -52,45 +151,51 @@ class DefinedSuperNode < BaseSuperNode
       handle :defined_super
 
       def compile
-        # insert method body to find super method
-        self.compile_dispatcher
-
-        wrap '((', ') != null ? "super" : nil)'
+        add_method(nil)
+        # will never come back null with method missing on
+        if compiler.method_missing?
+          wrap '(!(', '.$$stub) ? "super" : nil)'
+        else
+          # TODO: With method_missing support off, something breaks in runtime.js's chain
```

Comment:
This is interesting, we should either deprecate disabling MM or add some specs to ensure it "kinda" works, because I'm quite sure rubyspec relies on it and so does probably mspec (could be wrong tho).

@opal/core thoughts? (Personally I'm for deprecating it telling ppl to tell us if they actually disable it)